fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/services/ReferenceServiceImpl.java

```diff
@@ -302,6 +302,8 @@ public void updateReferences(@Nonnull final String txId, final FedoraId resource
             LOGGER.warn("Unable to update reference index for resource {} in transaction {}: {}",
                     resourceId.getFullId(), txId, e.getMessage());
             throw new RepositoryRuntimeException("Unable to update reference index", e);
+        } finally {
+            rdfStream.close();
```

Comment:
This will work, but it's slightly unexpected behavior. Generally speaking, streams are closed on the same level that they're opened on. For example,

```java
try (final stream = content.get()) {
    final RdfStream rdf = parseRdf(fedoraId, stream);
    this.referenceService.updateReferences(txId, fedoraId, null, rdf);
}
```

There are a variety of advantages to this approach, not the least of which is that it's clear when a stream is closed and you don't need to inspect every method that the stream is passed to to see if it closes it.

Reply:
So its going to end up looking like this
```java
try (final var stream = content.get()) {
    final RdfStream rdf = parseRdf(fedoraId, stream);
    this.referenceService.updateReferences(txId, fedoraId, null, rdf);
} catch (final IOException e) {
    LOGGER.warn("Content stream for {} closed prematurely, inbound references skipped.",
        fedoraId.getFullId());
    throw new RepositoryRuntimeException(e.getMessage(), e);
}
```

(I think) because its already in a `try {}` enclosure. Do I also need to close the RdfStream?

Reply:
Yes, that looks correct.

I poked around in the code, and it looks like what's going on is the InputStream is read into an RDF model object and then that model object is turned into an RdfStream. The RdfStream has no relation to the InputStream. It is a stream in the Java 8 Streaming since, and very likely does not need to be closed. Streams can be closed, but, unless the Stream is IO related, it is usually not necessary.